interfaces/builtin/fwupd.go

```diff
@@ -36,6 +36,7 @@ const fwupdBaseDeclarationSlots = `
     allow-installation:
       slot-snap-type:
         - app
+        - core
     deny-connection: true
```

Comment:
This should be changed to:
```
deny-connection:
  on-classic: false
```

Reply:
This would also affect the slot provided by the `uefi-fw-tools` snap when running on classic distros.  It probably doesn't matter much in the long run since the slot won't autoconnect, but it is a minor weakening of the policy,

Reply:
I'm not sure why we would want this change either,  there are even more reasons on classic not to want to connect to this one??

Reply:
I guess the question is whether a store assertion can change the snap declaration for the virtual `system` snap.  If that's not possible, then the base declaration needs to be something that will allow connection.

That means we'd need something in the plug side base declaration to control access.

Reply:
Yes, my idea would be that you would always need a plug side snap-declaration to connect this.

Reply:
ping @jdstrand

Reply:
The pattern for implicitClassic slots that may also be provided by an app has been:
```
    allow-installation:
      slot-snap-type:
        - app
        - core
    deny-auto-connection: true
    deny-connection:
      on-classic: false
```

The reasoning behind this is that when an app is providing the slot (ie a service for other snaps to plugs), we want to use deny-connection with deny-auto-connection since we don't want a user to even be able to manually connect without a snap declaration. For core, we trust the host enough to allow a manual connection by the user.

Reply:
> The pattern for implicitClassic slots that may also be provided by an app has been:
> 
> ```
>     allow-installation:
>       slot-snap-type:
>         - app
>         - core
>     deny-auto-connection: true
>     deny-connection:
>       on-classic: false
> ```
> 
> The reasoning behind this is that when an app is providing the slot (ie a service for other snaps to plugs), we want to use deny-connection with deny-auto-connection since we don't want a user to even be able to manually connect without a snap declaration. For core, we trust the host enough to allow a manual connection by the user.

However, you are right to point out that what I recommended is weakening the policy and the pattern should probably be changed to:
```
    allow-installation:
      slot-snap-type:
        - app
        - core
    deny-auto-connection: true
    deny-connection:
      slot-snap-type:
      - app
```

@pedronis - thoughts?

Reply:
@jdstrand we need to chat, there is some confusion here

Reply:
Note, historical context is that before we had experience with these, the pattern was developed because it was figured that say, bluez would either be provided as a snap on UC or from distro packaging on classic, but not a snap to replace distro packaging. This last case is still evolving since in most cases there isn't a straightforward way to fully replace the distro service with a snap considering all possible host integration points.

Reply:
@jhenstridge - ok, @pedronis and I discussed this and decided that regardless of historical context and existing patterns, fwupd is a new pattern where we do want `deny-connection: true` since it provides a lot of privilege to the connecting snap and we want to disallow even manual connection without a snap declaration.

Sorry for the flip/flop on this, but the PR isn't green need to rerun the tests anyway.

Reply:
I've reverted the changes to the base declaration now.

The `google:ubuntu-18.04-64:tests/main/mount-ns:inherit` failure in the last run seems to be seeing a stale mount for `/run/user/1234`.  The tests in this PR don't deal with the user session, so it looks to be unrelated.  Hopefully it won't fail in the current run.